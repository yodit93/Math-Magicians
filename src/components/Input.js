const Input = () => (
  <div className="grid">
    <p className="item">AC</p>
    <p className="item">+/-</p>
    <p className="item">%</p>
    <p className="item division color">รท</p>
    <p className="item">7</p>
    <p className="item">8</p>
    <p className="item">9</p>
    <p className="item color">x</p>
    <p className="item">4</p>
    <p className="item">5</p>
    <p className="item">6</p>
    <p className="item color">-</p>
    <p className="item">1</p>
    <p className="item">2</p>
    <p className="item">3</p>
    <p className="item color">+</p>
    <p className="item item0">0</p>
    <p className="item">.</p>
    <p className="item color">=</p>
  </div>
);

export default Input;
