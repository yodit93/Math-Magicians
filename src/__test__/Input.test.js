import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Input from '../components/Input';

describe('Input componenet', () => {
  let btn = '';
  const onHandleClick = (e) => {
    btn = e.target.textContent;
  };
  test('should render the Input component', () => {
    const tree = renderer.create(<Input onHandleClick={onHandleClick} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should have button with the text "AC" ', () => {
    render(<Input onHandleClick={onHandleClick} />);
    const buttonElement = screen.getByText(/AC/i);
    expect(buttonElement).toBeInTheDocument();
  });

  test('should have more than 10 buttons', () => {
    render(<Input onHandleClick={onHandleClick} />);
    const buttonElements = screen.getAllByRole('button');
    expect(buttonElements.length).toBeGreaterThan(10);
  });

  test('btn should equal to "Ac"', () => {
    render(<Input onHandleClick={onHandleClick} />);
    const buttonElement = screen.getByText('AC');
    fireEvent.click(buttonElement);
    expect(btn).toBe('AC');
  });

  test('btn should equal to "1"', () => {
    render(<Input onHandleClick={onHandleClick} />);
    const buttonElement = screen.getByText('1');
    fireEvent.click(buttonElement);
    expect(btn).toBe('1');
  });

  test('btn should equal to "%"', () => {
    render(<Input onHandleClick={onHandleClick} />);
    const buttonElement = screen.getByText('%');
    fireEvent.click(buttonElement);
    expect(btn).toBe('%');
  });

  test('btn should equal to "x"', () => {
    render(<Input onHandleClick={onHandleClick} />);
    const buttonElement = screen.getByText('x');
    fireEvent.click(buttonElement);
    expect(btn).toBe('x');
  });
});
