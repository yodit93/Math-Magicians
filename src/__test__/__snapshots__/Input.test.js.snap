// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input componenet should render the Input component 1`] = `
<div
  className="grid"
>
  <button
    className="item"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="item"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="item"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="item division color"
    onClick={[Function]}
    type="button"
  >
    รท
  </button>
  <button
    className="item"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="item"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="item"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="item color"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    className="item"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="item"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="item"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="item color"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="item"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="item"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="item"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="item color"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="item item0"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="item"
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="item color"
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;
